apps/meteor/app/livechat/server/lib/RoutingManager.ts

```diff
@@ -272,18 +270,19 @@ export const RoutingManager: Routing = {
 		return LivechatRooms.findOneById(rid);
 	},
 
-	async transferRoom(room, guest, transferData) {
-		logger.debug(`Transfering room ${room._id} by ${transferData.transferredBy._id}`);
-		if (transferData.departmentId) {
-			logger.debug(`Transfering room ${room._id} to department ${transferData.departmentId}`);
-			return forwardRoomToDepartment(room, guest, transferData);
-		}
+	async transferRoom(room, _guest, transferData) {
+		logger.debug(`Transferring room ${room._id} by ${transferData.transferredBy._id}`);
 
-		if (transferData.userId) {
-			logger.debug(`Transfering room ${room._id} to user ${transferData.userId}`);
+		if (transferData.userId || (transferData.userId && transferData.departmentId)) {
```

Comment:
Is this check needed here? If the transfer is happening to an agent, then userId is set. If that's `true`, then this side of the || won't have any effect (as the other side is already true).

I also see we prioritize agent forwarding now. Is that a requirement or an accident?